src/AutoRest.CSharp/Common/AutoRest/Plugins/GeneratedCodeWorkspace.cs

```diff
@@ -170,5 +171,236 @@ public async Task<CSharpCompilation> GetCompilationAsync()
         }
 
         public static bool IsGeneratedDocument(Document document) => document.Folders.Contains(GeneratedFolder);
+        private static bool IsMgmtRootDocument(Document document) => IsGeneratedDocument(document) && (!document.Name.Contains('/') || document.Name.Contains("Extensions/"));
```

Comment:
`Path.GetDirectoryName(document.Name) is "Extensions" or string.Empty`

Reply:
Unfortunately pattern matching can only apply to constants, and string.Empty is not a constant, therefore I have to use `""` instead of `string.Empty`.